src/snowflake/snowpark/_internal/analyzer/datatype_mapper.py

```diff
@@ -200,5 +204,5 @@ def to_sql_without_cast(value: Any, data_type: DataType) -> str:
     if value is None:
         return "NULL"
     if isinstance(data_type, StringType):
-        return f"""{value}"""
+        return f"'{value}'"
```

Comment:
This is a bug that we didn't realize before https://github.com/snowflakedb/thundersnow/blob/535a433f3157de6a43eea600797ab5c50cca5426/src/main/scala/com/snowflake/snowpark/internal/analyzer/DataTypeMapper.scala#L115

Reply:
Nope, it's possible that `value` here is numeric, but `data_type` is `StringType`.